docs/worker-versioning.md

```diff
@@ -0,0 +1,229 @@
+reachable, decommission the old Build ID.
+
+**Note for Temporal Server Operators:** Worker Versioning needs to be enabled from server
+before being able to run versioned workers or update the rules. To enable Worker Versioning
+set the following dynamic configs to `true`: `frontend.workerVersioningRuleAPIs`, 
+`frontend.workerVersioningWorkflowAPIs`. Both configs are per Task Queue.
+
+
+### Update Rules Using CLI
+You can use `temporal task-queue versioning` commands to update and read the Versioning 
+rules of a given Task Queue. Here are a few examples:
+
+**Add Assignment rule:** send 10% of new executions to Build ID "abc-123".
+```shell
+temporal task-queue versioning insert-assignment-rule --task-queue MY_TQ --build-id abc-123 --percentage 10
+```
+
+**Commit Build ID** to complete the rollout of "abc-123" and cleanup unnecessary rules.
+```shell
+temporal task-queue versioning commit-build-id --task-queue MY_TQ --build-id abc-123
+```
+
+**Add Redirect rule** from "abc-123" to "xyz-789":
+```shell
+temporal task-queue versioning add-redirect-rule --task-queue MY_TQ --source-build-id abc-123 --target-build-id xyz-789
+```
+
+**List Versioning rules**:
+```shell
+temporal task-queue versioning get-rules --task-queue MY_TQ
+```
+
+### Get Reachability Info via CLI
+You can use `temporal task-queue describe` command to get reachability status of a Build ID.
+Example: report reachability of Build ID "abc-123" on Task Queue MY_TQ:
+```shell
+temporal task-queue describe --task-queue MY_TQ --select-build-id abc-123 --report-reachability
+```
+
+The reachability status can be one of the following:
+- **REACHABLE:** Build ID may be used by new workflows or activities (base on versioning 
+rules), or there MAY be open workflows or backlogged activities assigned to it.
+- **CLOSED_WORKFLOWS_ONLY:** Build ID does not have open workflows and is not reachable
+by new workflows, but MAY have closed workflows within the namespace retention period. 
+Not applicable to activity-only task queues.
+- **UNREACHABLE:** Build ID is not used for new executions, nor it has been used by 
+any existing execution within the retention period.
+- **UNSPECIFIED:** Task reachability is not reported
+
+#### Caveats:
+- Task Reachability is eventually consistent; there may be a delay until it converges to the most
+accurate value but it is designed in a way to take the more conservative side until it converges.
+For example REACHABLE is more conservative than CLOSED_WORKFLOWS_ONLY.
+
+- Future activities who inherit their workflow's Build ID but not its Task Queue will not be
+accounted for reachability as server cannot know if they'll happen as they do not use
+assignment rules of their Task Queue. Same goes for Child Workflows or Continue-As-New Workflows
+who inherit the parent/previous workflow's Build ID but not its Task Queue. In those cases, make
+sure to query reachability for the parent/previous workflow's Task Queue as well.
+
+## Best Practices
```

Comment:
I think our customers would benefit from a  "Getting started" or walkthrough.  I suspect the reference guide you gave above is helpful for those diving into details, but a bit too abstract to get them started on building a real deployment solution.  In particular, it's not clear *when* to call all those commands and which ones are recommended.